src/webgpu/api/validation/createView.spec.ts

```diff
@@ -1,311 +1,318 @@
-    { format: 'r8unorm', _success: false },
-    { dimension: '2d-array', _success: true },
-    { dimension: '2d', _success: true },
-    { arrayLayerCount: 0, _success: true },
-    { arrayLayerCount: 1, _success: true },
-    { arrayLayerCount: 2, _success: false },
-    { mipLevelCount: MIP_LEVEL_COUNT, _success: true },
-    { mipLevelCount: 1, _success: true },
-  ] as const)
-  .fn(async t => {
-    const { format, dimension, arrayLayerCount, mipLevelCount, _success } = t.params;
+g.test('array_layers')
+  .desc(
+    `Views must have at least one layer, and must be within the layers of the base texture.
 
-    const texture = t.createTexture({ arrayLayerCount: 1 });
+  - arrayLayerCount=0 at various baseArrayLayer values
+  - Cases where baseArrayLayer+arrayLayerCount goes past the end of the texture
+  - Cases with baseArrayLayer or arrayLayerCount undefined (compares against reference defaulting impl)
+  `
+  )
+  .paramsSubcasesOnly(u =>
+    u
+      .combineWithParams([
+        { textureLayers: 1, textureLevels: 1 },
+        { textureLayers: kLayers, textureLevels: kLevels },
+      ])
+      .combine('view', [
+        {},
+        { baseArrayLayer: 0 },
+        { arrayLayerCount: 1 },
+        { baseArrayLayer: 0, arrayLayerCount: 1 },
+        { baseArrayLayer: 1, arrayLayerCount: 1 },
+        { baseArrayLayer: kLayers - 2, arrayLayerCount: 1 },
+        { baseArrayLayer: kLayers - 1, arrayLayerCount: 1 },
+        { baseArrayLayer: 0, arrayLayerCount: 2 },
+        { baseArrayLayer: 1, arrayLayerCount: 2 },
+        { baseArrayLayer: kLayers - 2, arrayLayerCount: 2 },
+        // arrayLayerCount == undefined means to use all remaining layers
+        { arrayLayerCount: undefined, baseArrayLayer: 0 },
+        { arrayLayerCount: undefined, baseArrayLayer: 1 },
+        { arrayLayerCount: undefined, baseArrayLayer: kLayers - 1 },
+        { arrayLayerCount: undefined, baseArrayLayer: kLayers },
+        // arrayLayerCount == 0 means zero array layers, which is never valid
+        { arrayLayerCount: 0, baseArrayLayer: 0 },
+        { arrayLayerCount: 0, baseArrayLayer: 1 },
+        { arrayLayerCount: 0, baseArrayLayer: kLayers - 1 },
+        { arrayLayerCount: 0, baseArrayLayer: kLayers },
+        // array layer range out of bounds
+        { baseArrayLayer: 0, arrayLayerCount: kLayers + 1 },
+        { baseArrayLayer: 1, arrayLayerCount: kLayers },
+        { baseArrayLayer: kLayers - 1, arrayLayerCount: 2 },
+        { baseArrayLayer: kLayers, arrayLayerCount: 1 },
+      ])
+  )
+  .fn(t => {
+    const { textureLayers, textureLevels, view } = t.params;
 
-    const descriptor = { format, dimension, arrayLayerCount, mipLevelCount };
+    const textureDescriptor: GPUTextureDescriptor = {
+      format: 'rgba8unorm',
+      size: [32, 32, textureLayers],
+      mipLevelCount: textureLevels,
+      usage: GPUTextureUsage.SAMPLED,
+    };
+    const reified = reifyTextureViewDescriptor(textureDescriptor, view);
```

Comment:
I feel like that you want to test `2d, 2d-array, cube, and cube-array` view types, but it seems to me that this function will always set viewDimension to `2d` for this test?

Reply:
Good point, maybe I can merge this with the dimension_layer test.